sklearn/feature_extraction/text.py

```diff
@@ -877,6 +883,7 @@ def transform(self, X):
                 "Iterable over raw text documents expected, string object received."
             )
 
+        self._check_transform_output_config()
```

Comment:
`HashingVectorizer` current disables `_SetOutputMixin`:

https://github.com/scikit-learn/scikit-learn/blob/15012376eaa3fda862d1f26c67ef4ae90315d71b/sklearn/feature_extraction/text.py#L568-L570

The original goal was to not define `set_output` or wrap `transform` if the estimator only outputs sparse data. In the design, I wanted a way to "disable everything" just in case a third party library ran into issues.

The alternatives to this PR are:

1. Have `_SetOutputMixin` wrap `transform` all the time, which will error if the output data is sparse.
2. Use Panda's DataFrame for sparse data and have a warning about the inefficiency.
3. Subclass SciPy sparse as described in [SLEP012- InputArray](https://scikit-learn-enhancement-proposals.readthedocs.io/en/latest/slep012/proposal.html) and use that for Sparse output.

Reply:
To me the consistent approach here would be to always wrap `transform` (at least in scikit-learn) and raise if output is sparse, which is the current / proposed behavior anyway. But at least we wouldn't have to patch estimators which currently don't have their transform method wrapped.